src/operator/quantization/mkldnn/mkldnn_quantized_fully_connected.cc

```diff
@@ -0,0 +1,133 @@
+}
+
+void MKLDNNQuantizedFullyConnectedForward(const nnvm::NodeAttrs &attrs,
+                                          const OpContext &ctx,
+                                          const std::vector<NDArray> &in_data,
+                                          const std::vector<OpReqType> &req,
+                                          const std::vector<NDArray> &out_data) {
+  TmpMemMgr::Get()->Init(ctx.requested[fullc::kTempSpace]);
+  FullyConnectedParam param = nnvm::get<FullyConnectedParam>(attrs.parsed);
+  const size_t num_inputs = param.no_bias ? 2 : 3;
+
+  CHECK_EQ(in_data.size(), static_cast<size_t>(num_inputs * 3));
+  CHECK_EQ(out_data.size(), 3U);
+
+  NDArray data = in_data[fullc::kData];
+  NDArray weight = in_data[fullc::kWeight];
+  const TShape &ishape = data.shape();
+
+  CHECK(data.dtype() == mshadow::kUint8)
+    << "MKLDNNQuantizedFullyConnected Op only supports uint8 for now, but got "
+    << mxnet::op::type_string(data.dtype());
+
+  if (ishape.ndim() != 2) {
+    CHECK(param.flatten)
+      << "QuantizedFullyConnected Op only supports flatten=true when ishape.ndim()!=2 for now.";
+    data = data.MKLDNNDataReshape(Shape2(ishape[0], ishape.ProdShape(1, ishape.ndim())));
+  }
+
+  const float min_data =
+    in_data[num_inputs + quantized_fc_enum::kDataMin].data().dptr<float>()[0];
+  const float max_data =
+    in_data[num_inputs + quantized_fc_enum::kDataMax].data().dptr<float>()[0];
+  const float min_weight =
+    in_data[num_inputs + quantized_fc_enum::kWeightMin].data().dptr<float>()[0];
+  const float max_weight =
+    in_data[num_inputs + quantized_fc_enum::kWeightMax].data().dptr<float>()[0];
+  float *min_output_ptr = out_data[quantized_fc_enum::kOutMin].data().dptr<float>();
+  float *max_output_ptr = out_data[quantized_fc_enum::kOutMax].data().dptr<float>();
+
+  auto data_range = (data.dtype() == mshadow::kInt8) ? kInt8Range : kUint8Range;
+  float data_scale = data_range / MaxAbs(min_data, max_data);
+  float weight_scale = kInt8Range / MaxAbs(min_weight, max_weight);
+
+  NDArray quantized_bias;
+  if (!param.no_bias) {
+    NDArray bias = in_data[fullc::kBias];
+    float min_bias = in_data[num_inputs + quantized_fc_enum::kBiasMin].data().dptr<float>()[0];
+    float max_bias = in_data[num_inputs + quantized_fc_enum::kBiasMax].data().dptr<float>()[0];
+    float bias_int32_rescale = data_scale * weight_scale * MaxAbs(min_bias, max_bias) / kInt8Range;
+
+    quantized_bias = NDArray(bias.storage_type(), bias.shape(),
+                             bias.ctx(), true, mshadow::kInt32);
+    int8_t *bias_ptr = bias.data().dptr<int8_t>();
+    int32_t *quantized_bias_ptr = quantized_bias.data().dptr<int32_t>();
+    size_t bias_size = bias.shape().Size();
+    #pragma omp parallel for num_threads(engine::OpenMP::Get()->GetRecommendedOMPThreadCount())
+    for (size_t i = 0; i < bias_size; ++i) {
+      quantized_bias_ptr[i] = bias_ptr[i] * bias_int32_rescale;
+    }
```

Comment:
Recommend to use mkldnn reorder instead.

Reply:
Will mkldnn reorder always better than this way (reorder may introduce overhead)? since bias is usually not a big array.